chalice/app.py

```diff
@@ -232,6 +232,11 @@ def get_access_control_headers(self):
 
         return headers
 
+    def __eq__(self, other):
```

Comment:
I wonder if `__eq__` or an identity check would be better here. The equality check is sufficient for validating that any configuration that you can setup in Chalice will be compatible with API Gateways handling of CORS preflight requests. However we could take it a step further and do identity checks on the CORSConfig objects, which would give the benefit of forcing users to design their APIs in a way that is logically consistent with API Gateway.

For example consider the following chalice app using this branch:
```python
from chalice import Chalice, CORSConfig

app = Chalice(app_name='foobar')

cors_a = CORSConfig()
cors_b = CORSConfig()

@app.route('/a', methods=['POST'], cors=cors_a)
def route_a_post():
    pass


@app.route('/a', methods=['GET'], cors=cors_a)
def route_a_get():
    pass


@app.route('/b', methods=['POST'], cors=cors_a)
def route_b_post():
    pass


@app.route('/b', methods=['GET'], cors=cors_b)
def route_b_get():
    pass
```

We are logically associating `cors_a` with route `/a` and `cors_b` with `/b`, since that is what API Gateway requires. But we made a typo and put `cors_a` on the `route_b_post` handler. With `__eq__` validation this will pass the validation step and get deployed and everything will work until `cors_a` and `cors_b` diverge. There error message that you get is pretty clear so it should be quick to correct the symptom, but maybe your design didn't take into account this limitation and now you are stuck redesigning some portion of your app's cors handling. 

With identity checking the above example code would throw a validation error and raise this issue with the designer earlier on so that they can plan around it. If they need more control over how the CORS requests are handled, a separate handler can be created that accepts OPTIONS requests and checks the `Access-Control-Request-Method` header to figure out which method is about to be called, and set the CORS headers in that response individually.

So its a relatively minor point but in my opinion identity check would be slightly better. We could change the `cors=True` argument to pass in a pre-created CORSConfig object (or cache it the first time it is created) so that it would pass identity checks.